autorest/azure/async.go

```diff
@@ -150,14 +155,13 @@ func (f Future) GetPollingDelay() (time.Duration, bool) {
 // running operation has completed, the provided context is cancelled, or the client's
 // polling duration has been exceeded.  It will retry failed polling attempts based on
 // the retry value defined in the client up to the maximum retry attempts.
-func (f Future) WaitForCompletion(ctx context.Context, client autorest.Client) error {
+func (f *Future) WaitForCompletion(ctx context.Context, client autorest.Client) error {
```

Comment:
It would be a breaking change though if the type is returned from a func and you don't capture the result in a local.  :(  https://play.golang.org/p/Dl3kO09oWUZ
Unfortunately this change is required as WaitForCompletion() calls methods that update the Future's state.
Tangentially I also can't help but wonder if this type is really safe from races when passed by value.  I need to dig into this some more.

Reply:
Good catch!